google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java

```diff
@@ -0,0 +1,219 @@
+                .set("bytesArray")
+                .toBytesArray(
+                    Arrays.asList(ByteArray.copyFrom("bytes1"), ByteArray.copyFrom("bytes2")))
+                .set("timestampArray")
+                .toTimestampArray(
+                    Arrays.asList(
+                        Timestamp.ofTimeSecondsAndNanos(10, 0),
+                        Timestamp.ofTimeSecondsAndNanos(20, 0)))
+                .set("dateArray")
+                .toDateArray(
+                    Arrays.asList(
+                        Date.fromYearMonthDay(2021, 2, 3), Date.fromYearMonthDay(2021, 3, 4)))
+                .build()));
+
+    try (ResultSet resultSet =
+        databaseClient
+            .singleUse()
+            .executeQuery(Statement.of("SELECT * FROM " + TABLE_NAME + " WHERE Id = 1"))) {
+      resultSet.next();
+
+      assertEquals(Value.int64(1L), resultSet.getValue("Id"));
+      assertEquals(Value.bool(true), resultSet.getValue("bool"));
+      assertEquals(Value.int64(10L), resultSet.getValue("int64"));
+      assertEquals(20D, resultSet.getValue("float64").getFloat64(), 1e-15);
+      assertEquals(Value.numeric(new BigDecimal("30")), resultSet.getValue("numeric"));
+      assertEquals(Value.string("stringValue"), resultSet.getValue("string"));
+      assertEquals(Value.bytes(ByteArray.copyFrom("bytesValue")), resultSet.getValue("bytes"));
+      assertEquals(
+          Value.timestamp(Timestamp.ofTimeSecondsAndNanos(1, 0)), resultSet.getValue("timestamp"));
+      assertEquals(Value.date(Date.fromYearMonthDay(2021, 1, 2)), resultSet.getValue("date"));
+      assertEquals(Value.boolArray(new boolean[] {false, true}), resultSet.getValue("boolArray"));
+      assertEquals(Value.int64Array(new long[] {100L, 200L}), resultSet.getValue("int64Array"));
+      assertArrayEquals(
+          new double[] {1000D, 2000D},
+          Doubles.toArray(resultSet.getValue("float64Array").getFloat64Array()),
+          1e-15);
+      assertEquals(
+          Value.numericArray(Arrays.asList(new BigDecimal("10000"), new BigDecimal("20000"))),
+          resultSet.getValue("numericArray"));
+      assertEquals(
+          Value.stringArray(Arrays.asList("string1", "string2")),
+          resultSet.getValue("stringArray"));
+      assertEquals(
+          Value.bytesArray(
+              Arrays.asList(ByteArray.copyFrom("bytes1"), ByteArray.copyFrom("bytes2"))),
+          resultSet.getValue("bytesArray"));
+      assertEquals(
+          Value.timestampArray(
+              Arrays.asList(
+                  Timestamp.ofTimeSecondsAndNanos(10, 0), Timestamp.ofTimeSecondsAndNanos(20, 0))),
+          resultSet.getValue("timestampArray"));
+      assertEquals(
+          Value.dateArray(
+              Arrays.asList(Date.fromYearMonthDay(2021, 2, 3), Date.fromYearMonthDay(2021, 3, 4))),
+          resultSet.getValue("dateArray"));
+    }
+  }
+
+  @Test
+  public void testReadNullInArrays() {
```

Comment:
I think it would be good to add null values for `INT64` and `FLOAT64` arrays as well in this test. Arrays of all types can always contain null values, regardless whether the column itself is nullable, and regardless of the array element type.

Reply:
These throw a `NPE` as they do if you use a `resultSet.getLongArray(...)` or `resultSet.getDoubleArray(...)`. I could test the exception if you feel like we should.

Reply:
No, in that case I actually think we should change the implementation. It is logical that `resultSet.getDoubleArray(..)` throws an `NPE`. I don't think it is logical that `resultSet.getValue("some-float64-array-col")` should ever throw an `NPE` because one of the elements is `null`.

Reply:
Yes, I think we should change that as well now that we have a chance. `Value` also has an `isNull()` method, so it would make sense to let `resultSet.getValue(...)` return a `Value` instance when the column is null. The user can then check whether the value is actually null by calling `Value#isNull()`.

Reply:
Awesome, changed the implementation to allow for that. If you could re-review, I'd appreciate it.